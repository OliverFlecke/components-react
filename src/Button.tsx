import React from 'react';

export interface ButtonProps
	extends React.ButtonHTMLAttributes<HTMLButtonElement> {
	buttonType?: ButtonType;
}

type ButtonType =
	| 'Primary'
	| 'Secondary'
	| 'Success'
	| 'Danger'
	| 'Warning'
	| 'Link'
	| 'Transparent';

export const Button: React.FC<ButtonProps> = (props: ButtonProps) => {
	const { children, buttonType, disabled } = props;

	return (
		<button
			className={`btn ${disabled ? '' : getColor(buttonType)}`}
			{...props}
		>
			{children}
		</button>
	);
};

Button.defaultProps = {
	buttonType: 'Primary',
};

function getColor(type?: ButtonType) {
	switch (type) {
		case 'Link':
			return 'btn-link';
		case 'Warning':
			return 'btn-warning';
		case 'Danger':
			return 'btn-danger';
		case 'Transparent':
			return 'btn-transparent';
		case 'Success':
			return 'btn-success';
		case 'Secondary':
			return 'btn-secondary';
		case 'Primary':
		default:
			return 'btn-primary';
	}
}
