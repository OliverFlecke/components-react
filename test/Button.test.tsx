import React from 'react';
import * as ReactDOM from 'react-dom';
import Button, { ButtonType } from '../src/Button';
import renderer from 'react-test-renderer';

describe('Button', () => {
	test('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<Button />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	test('render disabled button', () => {
		const div = document.createElement('div');
		ReactDOM.render(<Button disabled />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	test.each([
		'Primary',
		'Secondary',
		'Success',
		'Transparent',
		'Warning',
		'Danger',
		'Link',
	] as ButtonType[])('render button types', (type: ButtonType) => {
		const div = document.createElement('div');
		ReactDOM.render(<Button buttonType={type} />, div);
		ReactDOM.unmountComponentAtNode(div);
	});
});

describe('Button snapshots', () => {
	test.each([
		'Primary',
		'Secondary',
		'Success',
		'Transparent',
		'Warning',
		'Danger',
		'Link',
	] as ButtonType[])('renders correctly', (type: ButtonType) => {
		const tree = renderer
			.create(<Button buttonType={type}>Click me!</Button>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
