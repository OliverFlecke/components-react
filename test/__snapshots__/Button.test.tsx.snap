// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshots renders correctly 1`] = `
<button
  className="btn btn-primary"
>
  Click me!
</button>
`;

exports[`Button snapshots renders correctly 2`] = `
<button
  className="btn btn-secondary"
>
  Click me!
</button>
`;

exports[`Button snapshots renders correctly 3`] = `
<button
  className="btn btn-success"
>
  Click me!
</button>
`;

exports[`Button snapshots renders correctly 4`] = `
<button
  className="btn btn-transparent"
>
  Click me!
</button>
`;

exports[`Button snapshots renders correctly 5`] = `
<button
  className="btn btn-warning"
>
  Click me!
</button>
`;

exports[`Button snapshots renders correctly 6`] = `
<button
  className="btn btn-danger"
>
  Click me!
</button>
`;

exports[`Button snapshots renders correctly 7`] = `
<button
  className="btn btn-link"
>
  Click me!
</button>
`;
